cpp/src/arrow/builder.h

```diff
@@ -730,7 +730,14 @@ class ARROW_EXPORT FixedSizeBinaryBuilder : public ArrayBuilder {
   FixedSizeBinaryBuilder(const std::shared_ptr<DataType>& type,
                          MemoryPool* pool ARROW_MEMORY_POOL_DEFAULT);
 
-  Status Append(const uint8_t* value);
+  Status Append(const uint8_t* value) {
```

Comment:
I understand the performance benefits of moving this to the header but this also reminds me that we should have a look at our build process. This will inline the code in more places but also makes binaries a bit larger, if we would build using `-fLTO` we should be able to get rid of those duplications while having even a bit better performance at the same time. 

The change here is probably not significant at all but all the small ones add up over time.

Reply:
The downside with LTO is inflated build times (the linking step becomes much longer, and it isn't parallelized). Also IIRC some compilers have various bugs or limitations with LTO.